pkgs/applications/video/obs-studio/plugins/obs-backgroundremoval/default.nix

```diff
@@ -2,44 +2,52 @@
 , stdenv
 , fetchFromGitHub
 , cmake
+, ninja
 , obs-studio
 , onnxruntime
 , opencv
+, qt6
+, curl
 }:
 
 stdenv.mkDerivation rec {
   pname = "obs-backgroundremoval";
-  version = "0.5.16";
+  version = "1.1.13";
 
   src = fetchFromGitHub {
-    owner = "royshil";
+    owner = "occ-ai";
     repo = "obs-backgroundremoval";
-    rev = "v${version}";
-    hash = "sha256-E+pm/Ma6dZTYlX3DpB49ynTETsRS2TBqgHSCijl/Txc=";
+    rev = "${version}";
+    hash = "sha256-QoC9/HkwOXMoFNvcOxQkGCLLAJmsja801LKCNT9O9T0=";
   };
 
-  nativeBuildInputs = [ cmake ];
-  buildInputs = [ obs-studio onnxruntime opencv ];
+  nativeBuildInputs = [ cmake ninja ];
+  buildInputs = [ obs-studio onnxruntime opencv qt6.qtbase curl ];
 
   dontWrapQtApps = true;
 
   cmakeFlags = [
+    "--preset linux-x86_64"
+    "-DCMAKE_MODULE_PATH:PATH=/build/source/cmake"
     "-DUSE_SYSTEM_ONNXRUNTIME=ON"
     "-DUSE_SYSTEM_OPENCV=ON"
+    "-DDISABLE_ONNXRUNTIME_GPU=ON"
   ];
 
-  postInstall = ''
-    mkdir $out/lib $out/share
-    mv $out/obs-plugins/64bit $out/lib/obs-plugins
-    rm -rf $out/obs-plugins
-    mv $out/data $out/share/obs
+  buildPhase = ''
+     cd ..
+     cmake --build build_x86_64 --parallel
+  '';
+
+  installPhase = ''
+    cmake --install build_x86_64 --prefix "$out"
   '';
```

Comment:
Can't we use cmakeFlags and the normal phases?

Reply:
cmakeFlags will apply to the normal buildPhase, right?
How to specify flags for installPhase?
Where is the documentation about this?

Reply:
I think the project is using pretty new cmake stuff we don't have explicit support for.

---


> cmakeFlags will apply to the normal buildPhase, right?

It applies to the configure phase https://github.com/NixOS/nixpkgs/blob/c5b25a6eb94539cdeac8971e05f71e12e9e7e8a3/pkgs/by-name/cm/cmake/setup-hook.sh#L141-L144

> How to specify flags for installPhase?

Apparently there is no hook for that.

>  Where is the documentation about this?

https://github.com/NixOS/nixpkgs/blob/master/doc/hooks/cmake.section.md

Reply:
Ok